startup_scripts/460_cables.py

```diff
@@ -0,0 +1,233 @@
+    """Returns name of a cable in format:
+    device_a interface_a <---> interface_b device_b
+    or for circuits:
+    circuit_a termination_a <---> termination_b circuit_b
+    """
+    cable_name = []
+
+    for is_side_b, termination in enumerate([termination_a, termination_b]):
+        try:
+            power_panel_id = getattr(termination[0], "power_panel_id", None)
+            if power_panel_id:
+                power_feed = PowerPanel.objects.get(id=power_panel_id)
+                segment = [f"{power_feed}", f"{termination[0]}"]
+            else:
+                segment = [f"{termination[0].device}", f"{termination[0]}"]
+        except AttributeError:
+            segment = [f"{termination[0].circuit.cid}", f"{termination[0]}"]
+
+        if is_side_b:
+            segment.reverse()
+
+        cable_name.append(" ".join(segment))
+
+    return " <---> ".join(cable_name)
+
+
+def check_interface_types(*args):
+    for termination in args:
+        try:
+            if termination.type in VIRTUAL_INTERFACES:
+                raise Exception(
+                    f"⚠️ Virtual interfaces are not supported for cabling. "
+                    f"Termination {termination.device} {termination} {termination.type}"
+                )
+        except AttributeError:
+            # CircuitTermination dosn't have a type field
+            pass
+
+def check_terminations_are_free(*args):
+    any_failed = False
+    for termination in args:
+        if termination.cable_id:
+            any_failed = True
+            print(f"⚠️ Termination {termination} is already occupied with cable #{termination.cable_id}")
+    if any_failed:
+        raise Exception(f"⚠️ At least one end of the cable is already occupied.")
+
+cables = load_yaml("/opt/netbox/initializers/cables.yml")
+
+if cables is None:
+    sys.exit()
+
+for params in cables:
+    params["termination_a_class"] = get_termination_class(params.get("termination_a_class"))
+    params["termination_b_class"] = get_termination_class(params.get("termination_b_class"))
+
+    term_a = get_termination_object(params, side="a")
+    term_b = get_termination_object(params, side="b")
+
+    check_interface_types(term_a, term_b)
+
+    term_a_ct = ContentType.objects.get_for_model(term_a)
+    term_b_ct = ContentType.objects.get_for_model(term_b)
+
+    types_ok, msg = check_termination_types(term_a_ct, term_b_ct)
+    cable_name = get_cable_name((term_a, term_a_ct), (term_b, term_b_ct))
+
+    if not types_ok:
+        print(f"⚠️ Invalid termination types for {cable_name}. {msg}")
+        continue
+
+    if cable_in_cables((term_a, term_a_ct), (term_b, term_b_ct)):
+        continue
+
+    check_terminations_are_free(term_a, term_b)
+
+    params["termination_a_id"] = term_a.id
+    params["termination_b_id"] = term_b.id
+    params["termination_a_type"] = term_a_ct
+    params["termination_b_type"] = term_b_ct
+
+    cable = Cable.objects.create(**params)
```

Comment:
I believe we check in other scripts whether the resource was actually created or not, and only then show the message, don't we? If so, could you make this script behave the same?

Reply:
With Cables the _get_or_create()_ method doesn't do any good. In this script if we didn't crash until line 231 then it will be created. If a resource is not created there will be an exception raised.